examples/llm/src/models/mosaic_gpt.py

```diff
@@ -19,76 +19,76 @@
 from composer.metrics.nlp import LanguageCrossEntropy, Perplexity
 from composer.models.base import ComposerModel
 from omegaconf import DictConfig
+from omegaconf import OmegaConf as om
+from transformers import PreTrainedModel
+from transformers.modeling_outputs import CausalLMOutput
 
 import examples.llm.src.models.layers.attention as attention
 import examples.llm.src.models.layers.gpt_blocks as gpt_blocks
+from examples.llm.src.models.configuration_mosaic_gpt import MosaicGPTConfig
 from examples.llm.src.models.param_init_fns import MODEL_INIT_REGISTRY
 
 
-class MosaicGPT(nn.Module):
+class MosaicGPT(PreTrainedModel):
+    config_class = MosaicGPTConfig
+    base_model_prefix = 'mosaic_gpt'
```

Comment:
I thought we were going to start moving away from this name... (cuz OpenAI has a copyright on the gpt name) 
@abhi-mosaic how does 'mosaic_llm' sound?
(this may be outside the scope of this PR)

Reply:
there are so many places to scrub the name "GPT" if thats what we want to do. but good callout

Reply:
labeling this as "outside the scope of this PR". @abhi-mosaic if we want to avoid the wrath of OpenAI/MS lawyers we'll eventually need to do this.

Reply:
Hanlin said this was a long ways off. We had heard a rumor that PubMedGPT got renamed because of the GPT trademark thing, but it turned out to be because of the `PubMed` part. So we have plenty of lead time before we need to deal with this (if at all, we'll see if the courts say its enforceable)